diff --git a/Bugzilla/Search.pm b/Bugzilla/Search.pm
index 4340025..8a96a0a 100644
--- a/Bugzilla/Search.pm
+++ b/Bugzilla/Search.pm
@@ -280,6 +280,8 @@ sub init {
         push(@supptables, "LEFT JOIN flagtypes ON flagtypes.id = flags.type_id");
     }
 
+    Bugzilla::Hook::process('buglist_supptables', { fields => \@fields, supptables => \@supptables });
+
     my $minvotes;
     if (defined $params->param('votes')) {
         my $c = trim($params->param('votes'));
@@ -339,6 +341,10 @@ sub init {
     # All fields that don't have a . in their name should be specifyable
     # in the URL directly.
     my @legal_fields = grep { $_->name !~ /\./ } Bugzilla->get_fields;
+
+    # Hook allows clause into other tables than 'bugs'.
+    Bugzilla::Hook::process('buglist_supp_legal_fields', { fields => \@legal_fields });
+
     if (!$user->is_timetracker) {
         foreach my $field (TIMETRACKING_FIELDS) {
             @legal_fields = grep { $_->name ne $field } @legal_fields;
